### PR TITLE
Fixed #52 - Incorrect behavior when generate module which already exists

### LINKED ISSUE #52: Incorrect behavior when generate module which already exists
When regenerate module with name already exists 
1) No any warnings about deletion my files (it possible I already have some code here)
2) Incorrect deletion from project, some files paths corrupted in _compiled sources_
SS

steps:

```
generamba gen RootNavigation  bl_viper_default_gt
```

and again 

```
generamba gen RootNavigation  bl_viper_default_gt
```

### PR DESCRIPTION
**Pull request by Beniamiiin**

### CODE REVIEW

File: generamba/lib/generamba/module_generator.rb [hunk 1]
```diff
@@ -5,88 +6,85 @@
 module Generamba
 
 	# Responsible for creating the whole code module using information from the CLI
-	class ModuleGenerator
+	class ModuleGenerator < Thor
```
Comment (etolstoy) [inline generamba/lib/generamba/module_generator.rb]: Why are you subclassing this class from Thor?

File: generamba/lib/generamba/module_generator.rb [hunk 2]
```diff
@@ -5,88 +6,85 @@
...
+		no_commands {
```
Comment (etolstoy) [inline generamba/lib/generamba/module_generator.rb]: And a related question: what is this for?

**Comment by etolstoy** [comment]
@Beniamiiin thanks for contributing, a really nice implementation! I have a couple of questions - you can see them in code review.

Maybe you want to implement also another part of the issue?
`1) No any warnings about deletion my files (it possible I already have some code here)`

File: generamba/lib/generamba/module_generator.rb (hunk 1 above)
Comment (Beniamiiin) [inline generamba/lib/generamba/module_generator.rb]: If I don't subclassing from Thor, I don't use `yes?` method, from Thor module. Correct me if I am wrong.

Comment (etolstoy) [inline generamba/lib/generamba/module_generator.rb]: Oh, I missed two lines with the `yes` method.

I don't like the idea of using `Thor` and its functions on the business logic layer (and `ModuleGenerator`) definitely belongs to this layer. We shouldn't mess presentation and business logic in one class.

I see two possible solutions:
- You can perform this check somewhere in the `gen_command.rb` (maybe by using some helper class)
- You can post a warning, that these files are not empty, and stop the module generation process. To force overwrite, you can add some flag to `generamba gen` command, like `generamba gen MyModule MyTemplate --force`.

What do you think?

Comment (Beniamiiin) [inline generamba/lib/generamba/module_generator.rb]: I think, perform this check in `gen_command.rb` right idea. And I noticed, that I don't sold this check, but only sold `yes` answer :( Then, I'll implementing this method in `xcodeproj_helper.rb` and move this check in `gen_command.rb`.

**Comment by Beniamiiin** [comment]
Yes, I am implementing this too.

File: generamba/lib/generamba/module_generator.rb (hunk 2 above)
Comment (Beniamiiin) [inline generamba/lib/generamba/module_generator.rb]: Without this method all methods will be available as commands.

> `#no_commands`
> All methods defined inside the given block are not added as commands.

File: generamba/lib/generamba/cli/gen_command.rb [hunk 3]
```diff
@@ -39,6 +40,17 @@ def gen(module_name, template_name)
       template = ModuleTemplate.new(template_name)
       code_module = CodeModule.new(module_name, module_description, rambafile, options)
 
+      project = XcodeprojHelper.obtain_project(code_module.xcodeproj_path)
+      module_group_already_exists = XcodeprojHelper.module_with_group_path_already_exists(project, code_module.module_group_path)
+
+      if (module_group_already_exists == true)
```
Comment (etolstoy) [inline generamba/lib/generamba/cli/gen_command.rb]: As far as I know, ruby style guidelines require not using brackets for if cases

File: generamba/lib/generamba/cli/gen_command.rb [hunk 4]
```diff
@@ -39,6 +40,17 @@ def gen(module_name, template_name)
...
+        replace_exists_module = yes?("#{module_name} module already exists. Replace? (yes/no)")
+      
+        if (replace_exists_module == false)
```
Comment (etolstoy) [inline generamba/lib/generamba/cli/gen_command.rb]: It's better to post a message in this case, something like `You chose not to overwrite your module. Try using another name.`

File: generamba/lib/generamba/helpers/xcodeproj_helper.rb [hunk 5]
```diff
@@ -36,27 +36,52 @@ def self.add_file_to_project_and_targets(project, targets, group_path, file_path
...
       group_names = group_names_from_group_path(group_path)
 
       final_group = project
 
       group_names.each do |group_name|
         next_group = final_group[group_name]
-        if (next_group == nil)
-          next_group = final_group.new_group(group_name, group_name)
+        if next_group == nil
+          if create_group_in_not_exists == false
```
Comment (etolstoy) [inline generamba/lib/generamba/helpers/xcodeproj_helper.rb]: typo, in=if

File: generamba/lib/generamba/cli/gen_command.rb (hunk 3 above)
Comment (Beniamiiin) [inline generamba/lib/generamba/cli/gen_command.rb]: I use brackets only this module, because in this module already used brackets.

**Comment by etolstoy** [comment]
@Beniamiiin just a couple of minor fixes - and we can merge it! I thought of releasing a new version today - but I'll be unavailable for the following 10 days, so I won't be able to make a hotfix if something goes wrong.

I think we can schedule a new version something near the 15th of January.

**Comment by etolstoy** [comment]
Thanks once again!

**Pull request merged**